libs/pageserver_api/src/models.rs

```diff
@@ -177,6 +177,15 @@ serde_with::serde_conv!(
     |value: String| -> Result<_, humantime::TimestampError> { humantime::parse_rfc3339(&value) }
 );
 
+impl LsnLease {
+    pub const DEFAULT_LENGTH: Duration = Duration::from_secs(10 * 60);
+
+    /// Checks whether the lease is expired.
+    pub fn is_expired(&self) -> bool {
+        SystemTime::now() > self.valid_until
```

Comment:
If you take `now` as an input to this function, then you can avoid reading the system time repeatedly when checking many leases in a loop.